driver-core/src/main/com/mongodb/client/model/expressions/DateExpression.java

```diff
@@ -20,5 +20,18 @@
  * Expresses a date value.
  */
 public interface DateExpression extends Expression {
+    IntegerExpression year();
+    IntegerExpression month();
+    IntegerExpression dayOfMonth();
+    IntegerExpression dayOfWeek();
+    IntegerExpression dayOfYear();
+    IntegerExpression hour();
+    IntegerExpression minute();
+    IntegerExpression second();
+    IntegerExpression week();
+    IntegerExpression millisecond();
+
+    StringExpression dateToString();
+    StringExpression dateToString(StringExpression format, StringExpression timezone);
```

Comment:
Thoughts on the `onNull` expression supported by $dateToString?  Are you thinking to require `ifNull`?

Reply:
That server option presupposes that a DateExpression can have a value of null, which our API forbids. A user who has an Expression of an unknown type (could be null, could be date) would write:
```
expression.isDateOr(of(defaultInstance)).dateToString(...)
```
The `exp.isDateOr(deflt)` (and `doc.getDate("fieldName", deflt)`) methods are intended to be guards against nulls, and will arrive in the ensuing two PRs. I am unsure whether `ifNull` will even be necessary/coherent in our API.

(Incidentally, this is similar to the approach with limit on filter.)